qualtran/surface_code/reference.py

```diff
@@ -0,0 +1,45 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Optional
+
+from attrs import frozen
+
+
+@frozen
+class Reference:
+    """A reference to a source material.
+
+    Attributes:
+        source: A link to a paper/article (e.g. https://link-to-paper).
```

Comment:
should we be more specific with this attribute? if it's supposed to be a link then `url` is likely more appropriate

Reply:
done